src/dash-table/derived/cell/contents.tsx

```diff
@@ -48,69 +45,76 @@ function getCellType(
+    ) {
+
+    }
 
-            const dropdown = dropdowns[rowIndex][columnIndex];
-            const handlers = cellEventHandlerProps(propsFn)(rowIndex, columnIndex);
+    get = memoizeOne((
+        activeCell: ActiveCell,
+        columns: VisibleColumns,
+        data: Data,
+        offset: IViewportOffset,
+        editable: boolean,
+        isFocused: boolean,
+        dropdowns: (DropdownValues | undefined)[][]
+    ): JSX.Element[][] => mapData(
+        (datum, rowIndex) => mapRow(
+            (column, columnIndex) => {
+                const active = isActiveCell(activeCell, rowIndex + offset.rows, columnIndex + offset.columns);
 
-            const isEditable = isCellEditable(editable, column.editable);
+                const dropdown = dropdowns[rowIndex][columnIndex];
 
-            const className = [
-                ...(active ? ['input-active'] : []),
-                ...(isFocused ? ['focused'] : ['unfocused']),
-                ...['dash-cell-value']
-            ].join(' ');
+                const isEditable = isCellEditable(editable, column.editable);
 
-            switch (getCellType(active, isEditable, dropdown, column.presentation)) {
-                case CellType.Dropdown:
-                    return (<CellDropdown
-                        key={`column-${columnIndex}`}
-                        active={active}
-                        clearable={column.clearable}
-                        dropdown={dropdown}
-                        onChange={handlers.onChange}
-                        value={datum[column.id]}
-                    />);
-                case CellType.Input:
-                    return (<CellInput
-                        key={`column-${columnIndex}`}
-                        active={active}
-                        className={className}
-                        focused={isFocused}
-                        onChange={handlers.onChange}
-                        onClick={handlers.onClick}
-                        onDoubleClick={handlers.onDoubleClick}
-                        onMouseUp={handlers.onMouseUp}
-                        onPaste={handlers.onPaste}
-                        type={column.type}
-                        value={datum[column.id]}
-                    />);
-                case CellType.Label:
-                default:
-                    return (<CellLabel
-                        className={className}
-                        key={`column-${columnIndex}`}
-                        onClick={handlers.onClick}
-                        onDoubleClick={handlers.onDoubleClick}
-                        value={datum[column.id]}
-                    />);
-            }
-        },
-        columns
-    ),
-    data
-);
+                const className = [
+                    ...(active ? ['input-active'] : []),
+                    ...(isFocused ? ['focused'] : ['unfocused']),
+                    ...['dash-cell-value']
+                ].join(' ');
```

Comment:
Non-blocking: that'll work, and you're just moving this code around... but seems a bit convoluted. Couldn't it be simplified to:
```js
const className = [
    ...(active ? ['input-active'] : []),
    isFocused ? 'focused' : 'unfocused',
    'dash-cell-value'
].join(' ');
```

Reply:
sure